### PR TITLE
check whether a number is palindrome or not

### PR DESCRIPTION
**Pull request by muthuannamalai12**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?
Yes
(Write your answer here.)

### Description
Added a C Program to check whether a number is palindrome or not
(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests
#1925 
(Write your answer here.)

### CODE REVIEW

**Comment by muthuannamalai12** [comment]
@srimani-programmer . Kindly review

File: C/other/Palindrome.c [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	  Example 1:
+	  
+	  Enter a number: 121                                                                                                                                                                                                                                   
+	  Reverse of 121 is 121                                                                                                          
+	  121 is palindrome
+	  
+	  Example 2:
+	  
+	  Enter a number: 123                                                                                                                                                                                                                                   
+	  Reverse of 121 is 321
```
Comment (kmrakash) [inline C/other/Palindrome.c]: Mistyped at line number 42 . It should be 123

Comment (muthuannamalai12) [reply C/other/Palindrome.c]: Sorry for the mistake. Changed it now. Kindly review it again

**Comment by muthuannamalai12** [comment]
I have changed it. Kindly review it @kmrakash

**Comment by muthuannamalai12** [comment]
@iamrajiv @HarshCasper @atarax665 Kindly review this

File: C/other/Palindrome.c [hunk 2]
```diff
@@ -0,0 +1,45 @@
+// C Program to check whether a number is Palindrome or not
```
Comment (atarax665) [inline C/other/Palindrome.c]: Add an introduction pitch of Palindrome number.

File: C/other/Palindrome.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+	    }
+	    else
+	    {
+	        printf("\n %d is not a palindrome",num);
+	    }
+	    getchar();
+	}
+	
+	/*
+		Sample input/output:
```
Comment (atarax665) [inline C/other/Palindrome.c]: Add time and space complexity of the program too.

File: C/other/Palindrome.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
+// C Program to check whether a number is Palindrome or not
+	
+	#include<stdio.h>
```
Comment (atarax665) [inline C/other/Palindrome.c]: Rectify the indentation.

**Review by atarax665 — CHANGES_REQUESTED** [review]
Format your code according to [C++ Style guide](https://github.com/TesseractCoding/NeoAlgo/blob/master/C/C_Style_guide.md).

**Comment by muthuannamalai12** [comment]
I have changed it Kindly review it again @atarax665

File: C/other/Palindrome.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+{
+	int rev = 0, rem = 0, n = 0, num = 0;
+	printf("\n Enter a number:\t");
+	scanf("%d", &n);
+	num = n;
+	printf("\n Reverse of %d is ", n);
+	while (n > 0)
+	{
+		rem = n % 10;
+		rev = rev *10 + rem;
```
Comment (atarax665) [inline C/other/Palindrome.c]:
```suggestion
		rev = rev * 10 + rem;
```

File: C/other/Palindrome.c [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+An integer is a palindrome if the reverse of that number is equal to the original number.
+Example of Palindromes are 11,44,101,121,
+*/
+
+#include <stdio.h>
+#include <unistd.h>
+
+void main()
+{
+	int rev = 0, rem = 0, n = 0, num = 0;
```
Comment (atarax665) [inline C/other/Palindrome.c]: No need to initialize num and n as 0, you can just declare it as a normal variable.

File: C/other/Palindrome.c [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+
+	getchar();
+}
+
+/*
+Sample Case:
+Example 1:
+Enter a number: 121
+Reverse of 121 is 121
+121 is a palindrome
```
Comment (atarax665) [inline C/other/Palindrome.c:42]: Add a blank line after every sample Input/Output example.

**Review by atarax665 — CHANGES_REQUESTED** [review]

**Comment by muthuannamalai12** [comment]
I have changed it Kindly review it again @atarax665

File: C/other/Palindrome.c [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+An integer is a palindrome if the reverse of that number is equal to the original number.
+Example of Palindromes are 11,44,101,121,
+*/
+
+#include <stdio.h>
+#include <unistd.h>
+
+void main()
+{
+	int rev = 0, rem = 0, n,num ;
```
Comment (atarax665) [inline C/other/Palindrome.c]:
```suggestion
	int rev = 0, rem = 0, n, num ;
```

**Review by atarax665 — APPROVED** [review]

File: C/other/Palindrome.c (hunk 8 above)
Comment (muthuannamalai12) [reply C/other/Palindrome.c]: I have changed it @atarax665

**Comment by muthuannamalai12** [comment]
@iamrajiv Please Review

**Review by iamrajiv — APPROVED** [review]

**Comment by muthuannamalai12** [comment]
@HarshCasper Can you please add DWOC and level label for this issue here on my PR.

**Comment by muthuannamalai12** [comment]
@HarshCasper @atarax665. Can you please add DWOC and level label for this issue here on my PR.

**Comment by muthuannamalai12** [comment]
@HarshCasper @atarax665 @anushkrishnav . Can you please add DWOC and level label for this issue here on my PR.

**Pull request merged**